pallets/parachain-staking/src/lib.rs

```diff
@@ -2558,14 +2594,13 @@ pub mod pallet {
 			let collators = Self::compute_top_candidates();
 			// snapshot exposure for round for weighting reward distribution
 			for account in collators.iter() {
-				let state = <CandidateState<T>>::get(&account)
+				let state = <CandidateSnapshot<T>>::get(&account)
 					.expect("all members of CandidateQ must be candidates");
 				collator_count += 1u32;
-				delegation_count += state.delegators.0.len() as u32;
-				let amount = state.total_counted;
+				delegation_count += state.delegations.len() as u32;
+				let amount = state.total;
 				total += amount;
-				let exposure: CollatorSnapshot<T::AccountId, BalanceOf<T>> = state.into();
-				<AtStake<T>>::insert(now, account, exposure);
+				<AtStake<T>>::insert(now, account, state);
```

Comment:
this should significantly reduce the PoV because we are no longer reading the entire CandidateState